<?xml version="1.0"?>
<robot name="/* robot_name */">

  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>

  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>

  <material name="green">
    <color rgba="0.5 1 0 1"/>
  </material>


    <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.2"/>
      </geometry>
      <material name="blue"/>
    </visual>
   </link>

  <link name="base_link2">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.2"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  
    <joint name="pitch" type="continuous">
    <parent link="base_link"/>
    <child link="base_link2"/>
    <axis xyz="-1.0 0.0 0.0"/>
    <origin xyz="0 0 0.1"/>
  </joint>

  <link name="base_link3">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.2"/>
      </geometry>
      <material name="blue"/>
    </visual>
    </link>

  <joint name="yaw" type="continuous">
    <parent link="base_link2"/>
    <child link="base_link3"/>
    <axis xyz="0.0 -1.0 0.0"/>
    <origin xyz="0 0 0.1"/>
  </joint>

  <link name="head">
    <visual>
      <geometry>
        <mesh filename="package://IMU_ADQ_pck/urdf/mesh/human-head-simplified.dae" scale="0.008 0.008 0.008"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.3"/>
      <material name="white"/>
    </visual>
  </link>


  <joint name="roll" type="continuous">
    <parent link="base_link3"/>
    <child link="head"/>
    <origin xyz="0 0 0.1"/>
    <axis xyz="0.0 0.0 -1.0"/>
  </joint>

  <!-- This is the start of the reference part!-->



  <link name="base_link2_ref">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.2"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  
    <joint name="pitch_ref" type="continuous">
    <parent link="base_link"/>
    <child link="base_link2_ref"/>
    <axis xyz="1.0 0.0 0.0"/>
    <origin xyz="0 0 0.1" rpy="0 0 0"/>
    <dynamics damping="2" friction="0.0"/>
  </joint>

  <link name="base_link3_ref">
    <visual>
      <geometry>
        <cylinder length="0.2" radius="0.2"/>
      </geometry>
      <material name="blue"/>
    </visual>
    </link>

  <joint name="yaw_ref" type="continuous">
    <parent link="base_link2_ref"/>
    <child link="base_link3_ref"/>
    <axis xyz="0.0 -1.0 0.0"/>
    <origin xyz="0 0 0.1" rpy="0 0 0"/>
    <dynamics damping="2" friction="0.0"/>
  </joint>

  <link name="head_ref">
    <visual>
      <geometry>
        <mesh filename="package://IMU_ADQ_pck/urdf/mesh/human-head-simplified.stl" scale="0.008 0.008 0.008"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.3"/>
      <material name="green"/>
    </visual>
    <inertial>
  16       <mass value="10"/>
  17       <inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/>
  18 </inertial>
  </link>


  <joint name="roll_ref" type="continuous">
    <parent link="base_link3_ref"/>
    <child link="head_ref"/>
    <origin xyz="0 0 0.1" rpy="0 0 0"/>
    <axis xyz="0.0 0.0 -1.0"/>
    <dynamics damping="2" friction="0.0"/>
  </joint>

</robot>
